lib/inline-imports.js

```diff
@@ -66,13 +62,28 @@ module.exports = function inlineImports(fileContents, options, visited = new Set
         },
         visited,
       );
-
-      result.push(inlined);
-
-      files[resolvedPath] = true;
+      callback(inlined, true, i + 1);
     } else {
-      result.push(line);
+      callback(line, false, i + 1);
     }
   }
+}
+
+function inlineImports(fileContents, options, visited = new Set()) {
+  const result = [];
+  _inlineImports(fileContents, options, visited, function (line) {
+    result.push(line);
+  });
   return result.join('\n');
+}
+
+module.exports = inlineImports;
+module.exports.lineToImports = function (fileContents, options, visited = new Set()) {
+  const result = new Map();
+  _inlineImports(fileContents, options, visited, function (imports, match, lineNumber) {
```

Comment:
nit-pick: i would find a generator easier to read here. I'm not sure its worth the change, but if it inspires you feel free to make the change.

```js
for (const line of inlineImports(fileContents, options)) {

}
```

Reply:
👍